README.md

```diff
@@ -79,6 +79,82 @@ In the example below, we add processor `fr.inria.gforge.spoon.processors.CountSt
 </dependencies>
 ```
 
+## How to pass properties to a processor? 
+
+Spoon allow to pass custom properties to the processor you want to use. 
+
+For passing properties, one must:
+
+1. Define a processor with annotated fields
+2. Add a configuration for passing the value to those fields
+
+For example you can create a Processor to change the name of a specific class like below. Note the usage of `Property` annotation for the fields.
+
+```java
+package my.app.pkg;
+
+import spoon.processing.AbstractProcessor;
+import spoon.processing.Property;
+import spoon.reflect.declaration.CtClass;
+
+public class ProcessorWithProperty extends AbstractProcessor<CtClass> {
+
+    @Property
+    String oldClassName;
+
+    @Property
+    String newClassName;
+
+    @Override
+    public void process(CtClass element) {
+        if (element.getSimpleName().equals(this.oldClassName)) {
+            element.setSimpleName(this.newClassName);
+        }
+    }
+}
+```
+
+Then you can use the following configuration to use your processor:
+
+```xml
+<plugin>
+    <groupId>fr.inria.gforge.spoon</groupId>
+    <artifactId>spoon-maven-plugin</artifactId>
+    <configuration>
+      <processors>
+        <processor>my.app.pkg.ProcessorWithProperty</processor>
+      </processors>
+      <processorProperties>
+        <processorProperty>
+          <name>my.app.pkg.ProcessorWithProperty</name>
+          <properties>
+            <property>
+              <name>oldClassName</name>
+              <value>App</value>
+            </property>
+            <property>
+              <name>newClassName</name>
+              <value>NewName</value>
+            </property>
+          </properties>
+        </processorProperty>
+      </processorProperties>
+    </configuration>
+</plugin>
+```
+Please note that you have to specify for which processor the properties should be used with the `name` attribute.
+Primitive types are supported as well as list and maps containing string and integers. List are created by separating values with comma:
+
+```xml
+<value>one,two,three,"value containing a, comma"</value>
```

Comment:
what about using JSON for the value? client code could then use any JSON lib to handle this.